### PR TITLE
Allow spans to be deleted

### PR DESCRIPTION
**Pull request by hatchan**
Allow deleting a specific span or all spans of a trace

NOTE: Support for deleting a batch of spans or having a cronjob that will delete old spans will be added in another PR.

### CODE REVIEW

File: fpx-lib/src/api/handlers/spans.rs [hunk 1]
```diff
@@ -101,3 +101,40 @@ impl From<DbError> for ApiServerError<SpanListError> {
...
+pub async fn span_delete_handler(
+    State(store): State<BoxedStore>,
+    Path((trace_id, span_id)): Path<(String, String)>,
+) -> Result<StatusCode, ApiServerError<SpanDeleteError>> {
+    let tx = store.start_readonly_transaction().await?;
+
+    hex::decode(&trace_id)
+        .map_err(|_| ApiServerError::ServiceError(SpanDeleteError::InvalidTraceId))?;
+    hex::decode(&span_id)
+        .map_err(|_| ApiServerError::ServiceError(SpanDeleteError::InvalidSpanId))?;
```
Comment (mellowagain) [inline fpx-lib/src/api/handlers/spans.rs:115]: would be cool to write a custom extractor so this wouldnt have to be repeated for every route we use those on

File: fpx-lib/src/api/handlers/spans.rs [hunk 2]
```diff
@@ -101,3 +101,40 @@ impl From<DbError> for ApiServerError<SpanListError> {
...
+
+    let _ = store.span_delete(&tx, &trace_id, &span_id).await?;
```
Comment (mellowagain) [inline fpx-lib/src/api/handlers/spans.rs]: is this marked `#[must_use]` that this is bound to `_`?

**Review by mellowagain — APPROVED** [review]

File: fpx-lib/src/api/handlers/spans.rs (hunk 1 above)
Comment (hatchan) [reply fpx-lib/src/api/handlers/spans.rs:115]: I'll add a type for it, that can do the conversion :+1:

File: fpx-lib/src/api/handlers/spans.rs (hunk 2 above)
Comment (hatchan) [reply fpx-lib/src/api/handlers/spans.rs]: No it isn't, will remove the assignment

File: fpx-lib/src/api/handlers/spans.rs (hunk 1 above)
Comment (hatchan) [reply fpx-lib/src/api/handlers/spans.rs:115]: https://linear.app/fiberplane/issue/FP-4039/create-typed-traceidspanid

**Review by stephlow — APPROVED** [review]

**Pull request merged**